resources/asset.rb

```diff
@@ -10,20 +10,23 @@
 property :owner, String
 property :group, String
 property :force, [true, false], default: false
+property :extract_to, String, required: [:extract]
 
 action :download do
-  chef_gem 'octokit' do
-    compile_time true
-  end
+  do_download
+end
 
-  unless asset.downloaded?(asset_path)
-    converge_by "downloading #{new_resource.file} to #{asset_path}" do
-      asset.download(
-        user: new_resource.github_user,
-        token: new_resource.github_token,
-        force: new_resource.force,
-        path: asset_path
-      )
+action :extract do
+  do_download
+
+  unless new_resource.force || ::File.exist?(new_resource.extract_to)
+    converge_by "extracting #{new_resource.file} to #{new_resource.extract_to}" do
```

Comment:
How come we need a converge_by here?

Reply:
I have no idea, I just copied it from the the other action :joy: